Makefile

```diff
@@ -10,3 +10,4 @@ check:
 
 clean:
 	rm -f *.o $(PROGS)
+	rm -f mmap-datafile
```

Comment:
Just thought of this -- we should follow the pattern we've setup here   define a new variable called `GENERATED` after `SCRIPTS` and set it to `:=mmap-datafile`.   Then adjust the `rm` invocation to `rm -f $(PROGS) $(GENERATED)`.  (We can do the `rm` all on one line and also get rid of the *.o since the compiler is generating programs all in one step, without intermediate .o object files.)